core/src/main/scala/com/rallyhealth/vapors/factfilter/extras/CaptureAllTimestamps.scala

```diff
@@ -0,0 +1,46 @@
+package com.rallyhealth.vapors.factfilter.extras
+
+import java.time.Instant
+
+import cats.Eval
+import cats.instances.list._
+import com.rallyhealth.vapors.core.algebra.Expr
+import com.rallyhealth.vapors.factfilter.data.{FactType, TypedFact}
+import com.rallyhealth.vapors.factfilter.dsl.CaptureP
+import com.rallyhealth.vapors.factfilter.evaluator.InterpretExprAsFunction.{Input, Output}
+
+object CaptureAllTimestamps extends CaptureAllTimestampsLowPriorityImplicits {
+
+  type Param = List[(FactType[_], Instant)]
+
+  implicit def captureTimestampsFromFacts[T : ExtractInstant, A]: CaptureP[List, TypedFact[T], A, Param] = {
+    new CaptureP.AsMonoid[List, TypedFact[T], A, Param] {
+
+      override protected def combineWithParent(
+        expr: Expr[List, TypedFact[T], A, Param],
+        input: Input[List, TypedFact[T]],
+        output: Output[A],
+        processedChildren: Param,
+      ): Eval[Param] = {
+        val timestamps = input.value.map(fact => (fact.typeInfo, ExtractInstant[T].extractValue(fact.value)))
+        Eval.now(timestamps)
+      }
+    }
+  }
```

Comment:
This is kind of redundant with the `CaptureTimeRange` example. I'm also not sure it should live in the main code base... maybe it should just be a test example until I have a good reason to expose it as a feature.

Reply:
Removed.